social_django/fields.py

```diff
@@ -7,8 +8,29 @@
 
 from social_core.utils import setting_name
 
-if getattr(settings, setting_name('POSTGRES_JSONFIELD'), False):
-    from django.contrib.postgres.fields import JSONField as JSONFieldBase
+POSTGRES_JSONFIELD = getattr(settings, setting_name('POSTGRES_JSONFIELD'), False)
+
+if POSTGRES_JSONFIELD:
+    warnings.warn(
+        'SOCIAL_AUTH_POSTGRES_JSONFIELD has been renamed to '
+        'SOCIAL_AUTH_JSONFIELD_ENABLED and will be removed in the next release.'
+    )
+    JSONFIELD_ENABLED = True
+else:
+    JSONFIELD_ENABLED = getattr(settings, setting_name('JSONFIELD_ENABLED'), False)
+
+if JSONFIELD_ENABLED:
+    JSONFIELD_CUSTOM = getattr(settings, setting_name('JSONFIELD_CUSTOM'), None)
+    if JSONFIELD_CUSTOM is not None:
+        try:
+            from django.utils.module_loading import import_string as import_module
+        except ImportError:
+            from importlib import import_module
+        JSONFieldBase = import_module(JSONFIELD_CUSTOM)
+    try:
+        from django.db.models import JSONField as JSONFieldBase
+    except ImportError:
+        from django.contrib.postgres.fields import JSONField as JSONFieldBase
```

Comment:
This will switch all existing django.contrib.postgres.fields.JSONField users to django.db.models.JSONField once they are on Django 3.1. Is this intended behaviour? Won't this cause some problems?

Reply:
That would raise an indication of some sort. I don't think there's an alternative / safe default in this instance, since we don't know the JSON Field the project is using.

They will be able to customize `JSONFIELD_CUSTOM`

Reply:
Sorry, I was not clearly describing the problematic case. 

The problem is with `POSTGRES_JSONFIELD` enabled. Before this change it was using `django.contrib.postgres.fields.JSONField`, but after this patch it changes to `django.db.models.JSONField` on Django 3.1+.

Reply:
I fixed two issues:
1. `django.contrib.postgres.fields.JSONField` is now the default option, unless an `ImportError` is raised. So there's no behavioral changes if the user upgrades. They can opt-in to the new field by setting `SOCIAL_AUTH_JSONFIELD_CUSTOM`.
2. If `SOCIAL_AUTH_JSONFIELD_CUSTOM` is enabled, that'll be the field used (before, it would override the custom field because no `else` was there)

Reply:
I think, this should be reverted to the original order. That is, first try to import from `django.db.models`, and if that fails then import from `django.contrib.postgres.fields`.

The `from django.contrib.postgres.fields import JSONField` import will never fail. It has to work to support historical migrations. The field is "removed" by having Django system checks fail when used in an actual model. This can be seen on [django master](https://github.com/django/django/blob/8fdb5a656a35dc9beef4cf2b79a782068263c7ec/django/contrib/postgres/fields/jsonb.py), which currently is the future Django 4.0.

Secondly, on Django 3.1+ the `django.contrib.postgres.fields.JSONField` is a subclass of `django.db.models.JSONField` adding nothing else than a system check warning. So everyone who previously used `django.contrib.postgres.fields.JSONField`, after upgrading to Django 3.1 immediately start using `django.db.models.JSONField`, and they only need to change the import location to avoid the warning.

Reply:
Ah, I didn't look into the Django code, but silently switching from `django.contrib.postgres.fields.JSONField` to `django.db.models.JSONField` seemed like an unwanted behaviour. If Django would do that anyway, then it's probably safe...

Reply:
@vytisb @nijel I will switch this around (and update the PR README, Changelog, etc) accordingly.

And still - in either case - if there's an edgecase, the user can always set `SOCIAL_AUTH_JSONFIELD_CUSTOM` to override.

So we'll have both 1. A sensible, intuitive default that matches Django's behavior and 2. An ability for the user to handle edgecases without forking

Reply:
@vtisb @nijel I updated this to default to `django.db.models.JSONField` if it exists, and also rebased.

Reply:
@tony I don't see the change. Maybe it got lost during rebase? Could you check again? Thanks.

Reply:
@vparitskiy 

Good eye! How is it now?

Reply:
@tony Looks good. Thanks!